torch/utils/cpp_extension.py

```diff
@@ -1675,7 +1713,7 @@ def object_file_path(source_file: str) -> str:
     elif IS_WINDOWS:
         ldflags = _nt_quote_args(ldflags)
 
-    ext = 'pyd' if IS_WINDOWS else 'so'
+    ext = 'pyd' if IS_WINDOWS else ('o' if is_standalone else 'so')
```

Comment:
The Windows branch here is wrong. pyd = shared library that can be loaded by Python on Windows. You almost definitely want .o here.

Reply:
Some common extension translations between Windows and Linux
```
pyd/dll = so
obj = o
lib = a
```

Reply:
By the way, the file extension of an executable is `.exe` in case if you need that.